[iOS] Shell/navigation page title view - respect the margin property

kubaflo opened the pull request:
<!-- Please let the below note in for people that find this PR -->
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

<!--
!!!!!!! MAIN IS THE ONLY ACTIVE BRANCH. MAKE SURE THIS PR IS TARGETING MAIN. !!!!!!! 
-->

### Description of Change

On iOS, the NavigationPage TitleView is currently overflowing, causing its content to be cut off.
This appears to be related to https://github.com/dotnet/maui/pull/20959, which removed the default margin. 

While the extra margin was not ideal and the change was meant to improve alignment, it introduced new layout issues (as discussed in https://github.com/dotnet/maui/issues/31688.

This PR refines that behavior:
- It replaces the custom CalculateUIEdgeInsets method with an override of AlignmentRectInsets that uses the IView.Margin property.
- By default, the native iOS gap is preserved to prevent layout regressions.
- Developers who want the TitleView aligned flush with the screen edges can now achieve this by applying a negative margin.
- This approach maintains compatibility with existing apps while providing more flexibility for developers who need precise control over TitleView alignment.

|Code|View|
|--|--|
|<img width="300" alt="Screenshot 2025-09-20 at 01 59 42" src="https://github.com/user-attachments/assets/5a317e53-ea64-443d-b509-092d97e52b50" />|<img width="300"  alt="Screenshot 2025-09-20 at 02 00 36" src="https://github.com/user-attachments/assets/8d060560-8890-4034-a121-5b85c5b83487" />|
|<img width="300" alt="Screenshot 2025-09-20 at 02 01 28" src="https://github.com/user-attachments/assets/2de071de-7798-4a13-83b3-7c8400d26aa6" />|<img width="300" alt="Screenshot 2025-09-20 at 02 02 22" src="https://github.com/user-attachments/assets/20088582-2be1-41c2-adc3-81bd391863a5" />|

### Issues Fixed

<!-- Please make sure that there is a bug logged for the issue being fixed. The bug should describe the problem and how to reproduce it. -->

Fixes https://github.com/dotnet/maui/issues/31688
Fixes https://github.com/dotnet/maui/pull/31681#issuecomment-3312115756

kubaflo committed:
[iOS] Shell/NavigationPage TitleView

Replaces the custom CalculateUIEdgeInsets method with an override of AlignmentRectInsets that leverages the IView.Margin property. This ensures the alignment insets reflect the view’s actual margins, allowing developers to align the TitleView directly with the screen edges.

PureWeen commented:
/rebase

PureWeen commented:
@sheiksyedm can you test this pr please

sheiksyedm commented:
@kubaflo  Please resave the latest iOS and Mac image from the CI run. 
<img width="1191" height="131" alt="image" src="https://github.com/user-attachments/assets/c78e1446-3092-4709-9a70-51e0ea12448d" />

PureWeen commented:
/azp run

Felix-Dev commented:
@kubaflo Where is that -20 Margin coming from to align the TitleView with the edges of the screen? Ideally, I as developer don't want to guess/experiment what the correct negative offset here would be to correctly align the the view to the edges.

If that -20 value is a value based on official data, could something like a resource be added to MAUI which developers could then reference to properly remove the default margin entirely?

kubaflo commented:
> @kubaflo Where is that -20 Margin coming from to align the TitleView with the edges of the screen? Ideally, I as developer don't want to guess/experiment what the correct negative offset here would be to correctly align the the view to the edges.
> 
> If that -20 value is a value based on official data, could something like a resource be added to MAUI which developers could then reference to properly remove the default margin entirely?

Yeah, that’s the tricky part - iOS doesn’t expose any public API (at least none that I know of) to reliably remove or control that default margin. To make it more complicated, the margin itself isn’t consistent: it can vary depending on screen size, device orientation, and even whether the “Back” text is shown in the navigation bar.

Because of that, if MAUI were to just hard-code the margin to 0, it would actually cause more layout issues than it solves. Without an official value from Apple or a stable API to query, the safest option for now is to preserve the native behavior and let developers apply their own negative margin when they want to override it.

kubaflo committed:
[iOS] Shell/NavigationPage TitleView

Replaces the custom CalculateUIEdgeInsets method with an override of AlignmentRectInsets that leverages the IView.Margin property. This ensures the alignment insets reflect the view’s actual margins, allowing developers to align the TitleView directly with the screen edges.

PureWeen commented:
/azp run
